stdlib/public/Distributed/DistributedActorSystem.swift

```diff
@@ -249,12 +243,37 @@ extension DistributedActorSystem {
     }
 
     do {
+      // Decode the invocation and pack arguments into the h-buffer
+      // TODO(distributed): decode the generics info
+      var argumentDecoder = invocation.makeArgumentDecoder()
+      var paramIdx = 0
+      for unsafeRawArgPointer in hargs {
+        guard paramIdx < paramCount else {
+          throw ExecuteDistributedTargetError(
+            message: "Unexpected attempt to decode more parameters than expected: \(paramIdx + 1)")
+        }
+        let paramType = paramTypes[paramIdx]
+        paramIdx += 1
+
+        // FIXME(distributed): func doDecode<Arg: SerializationRequirement>(_: Arg.Type) throws {
+        // FIXME:     but how would we call this...?
+        // FIXME:     > type 'Arg' constrained to non-protocol, non-class type 'Self.Invocation.SerializationRequirement'
```

Comment:
Maybe instead of having it be a requirement on `doDecode` we'd just generate some code to check conformance to `SerializationRequirement` while receiving the data for an invocation, so that we know that if we got an instance of it always has valid data?

Reply:
Discussed today: we'll have to keep this an ad-hoc requirement and call it from the accessor.